packages/common/http/src/xhr.ts

```diff
@@ -79,6 +79,12 @@ export class HttpXhrBackend implements HttpBackend {
       );
     }
 
+    if (req.keepalive && ngDevMode) {
+      console.warn(
+        `Angular detected that HttpClient keepalive with XHR does not function reliably. It is strongly recommended to enable the fetch API for proper keepalive support. To enable fetch, include \`withFetch()\` in the \`provideHttpClient()\` call at the application's root.`,
+      );
+    }
```

Comment:
```suggestion
    if (req.keepalive && ngDevMode) {
      console.warn(
        `Angular detected that a \`HttpClient\` request with the \`keepalive`\ option  was sent using XHR which does not support it. To use the \`keepalive\` option, enable fetch API support by passing \`withFetch()\` as argument to \`provideHttpClient()\`.,
      );
    }
```

Also ideally we would want that warning to be logged only once to prevent poluting the console. 

We have a similar behavior here: https://github.com/angular/angular/blob/cbd6ec8c6a3c632089cc8668d621aa597627af16/packages/common/http/src/transfer_cache.ts#L371-L388

Reply:
I see, so I should create a specific error code for this warning.
I guess I should add it to this errors.ts file?

`/**
 * The list of error codes used in runtime code of the `common/http` package.
 * Reserved error code range: 2800-2899.
 */
export const enum RuntimeErrorCode {
  MISSING_JSONP_MODULE = -2800,
  NOT_USING_FETCH_BACKEND_IN_SSR = 2801,
  HEADERS_ALTERED_BY_TRANSFER_CACHE = 2802,
  HTTP_ORIGIN_MAP_USED_IN_CLIENT = 2803,
  HTTP_ORIGIN_MAP_CONTAINS_PATH = 2804,
  CANNOT_SPECIFY_BOTH_FROM_STRING_AND_FROM_OBJECT = 2805,
  RESPONSE_IS_NOT_AN_ARRAY_BUFFER = 2806,
  RESPONSE_IS_NOT_A_BLOB = 2807,
  RESPONSE_IS_NOT_A_STRING = 2808,
  UNHANDLED_OBSERVE_TYPE = 2809,
  JSONP_WRONG_METHOD = 2810,
  JSONP_WRONG_RESPONSE_TYPE = 2811,
  JSONP_HEADERS_NOT_SUPPORTED = 2812,
}
`

Reply:
Update with formart code , documentation and warning message acording Angular format

Reply:
@JeanMeche  Are any additional changes needed from me for this to be approved? I'm willing to contribute.